subsys/bluetooth/audio/has.c

```diff
@@ -328,9 +305,97 @@ BT_CONN_CB_DEFINE(conn_cb) = {
 	.disconnected = disconnected,
 	.security_changed = security_changed,
 };
+
+static void notify_work_reschedule(struct has_client *client, enum has_notify notify,
+				   k_timeout_t delay)
+{
+	int err;
+
+	atomic_set_bit(client->notify, notify);
+
+	if (client->conn == NULL) {
+		return;
+	}
+
+	err = k_work_reschedule(&client->notify_work, delay);
+	if (err < 0) {
+		LOG_ERR("Failed to reschedule notification work err %d", err);
+	}
+}
+
+static void notify_work_handler(struct k_work *work)
+{
+	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
+	struct has_client *client = CONTAINER_OF(dwork, struct has_client, notify_work);
+	int err;
+
+#if defined(CONFIG_BT_HAS_FEATURES_NOTIFIABLE)
+	if (atomic_test_and_clear_bit(client->notify, NOTIFY_FEATURES) &&
+	    bt_gatt_is_subscribed(client->conn, FEATURES_ATTR, BT_GATT_CCC_NOTIFY)) {
+		err = bt_gatt_notify(client->conn, FEATURES_ATTR, &has.features,
+				     sizeof(has.features));
+		if (err == -ENOMEM) {
+			notify_work_reschedule(client, NOTIFY_FEATURES,
+					       K_USEC(BT_AUDIO_NOTIFY_RETRY_DELAY_US));
+		} else if (err < 0) {
+			LOG_ERR("Notify features err %d", err);
+		}
+	}
+#endif /* CONFIG_BT_HAS_FEATURES_NOTIFIABLE */
```

Comment:
Correct me if I am wrong, but it doens't seem like this block contains any data that isn't always available, in which case the `if defined` can be replaced with `if IS_ENABLED`